app/src/main/java/com/nutomic/syncthingandroid/service/SyncthingRunnable.java

```diff
@@ -317,44 +298,22 @@ public void killSyncthing() {
                 break;
             }
 
+            int exitCode;
+            String shellCmd;
             for (String syncthingPID : syncthingPIDs) {
-                killProcessId(syncthingPID, i > 0);
-            }
-        }
-    }
-
-    /**
-     * Kill a given process ID
-     *
-     * @param force Whether to use a SIGKILL.
-     */
-    private void killProcessId(String id, boolean force) {
-        Process kill = null;
-        DataOutputStream killOut = null;
-        try {
-            kill = Runtime.getRuntime().exec((mUseRoot) ? "su" : "sh");
-            killOut = new DataOutputStream(kill.getOutputStream());
-            if (!force) {
-                killOut.writeBytes("kill -SIGINT " + id + "\n");
-                killOut.writeBytes("sleep 1\n");
-            } else {
-                killOut.writeBytes("sleep 3\n");
-                killOut.writeBytes("kill -SIGKILL " + id + "\n");
-            }
-            killOut.writeBytes("exit\n");
-            killOut.flush();
-            kill.waitFor();
-            Log.i(TAG_KILL, "Killed Syncthing process "+id);
-        } catch (IOException | InterruptedException e) {
-            Log.w(TAG_KILL, "Failed to kill process id "+id, e);
-        } finally {
-            try {
-                if (killOut != null)
-                    killOut.close();
-            } catch (IOException e) {
-                Log.w(TAG_KILL, "Failed close the killOut stream", e);}
-            if (kill != null) {
-                kill.destroy();
+                if (i > 0) {
+                    // Force termination of the process by sending SIGKILL.
+                    shellCmd = "sleep 3; kill -SIGKILL " + syncthingPID + "\n";
+                } else {
+                    shellCmd = "kill -SIGINT " + syncthingPID + "; sleep 1\n";
+                }
+                exitCode = Util.runShellCommand(shellCmd, mUseRoot);
+                if (exitCode == 0) {
```

Comment:
If the process no longer exists, it will not be 0

Reply:
yes, according to my terminal emulator testing and source ( http://man7.org/linux/man-pages/man1/kill.1.html ) it will return exit code 1 then. It's called "failure" in the docs, so this code part is right logging "Failed to kill syncthing process ### exit code 1" then.